code/game/turfs/turf.dm

```diff
@@ -346,3 +344,9 @@
 				O.singularity_act()
 	ChangeTurf(/turf/space)
 	return(2)
+
+/turf/proc/can_have_cabling()
+	return !density
+
+/turf/proc/can_lay_cable()
+	return can_have_cabling() & !intact
```

Comment:
can_have_cabling() always returns 1 or 0, so that shouldn't matter but I can change it if you want.

return !density because only non dense turfs should be able to have cables (walls are dense, floors are not dense), you told me this yourself.